packages/office-ui-fabric-react/src/components/Panel/Panel.tsx

```diff
@@ -158,12 +158,18 @@ export class Panel extends BaseComponent<IPanelProps, IPanelState> {
                 ['ms-Panel--lg ' + styles.rootIsLarge]: type === PanelType.large || type === PanelType.largeFixed,
                 ['ms-Panel--fixed ' + styles.rootIsFixed]: type === PanelType.largeFixed,
                 ['ms-Panel--xl ' + styles.rootIsXLarge]: type === PanelType.extraLarge,
+                ['ms-Panel--custom ' + styles.rootIsCustom]: type === PanelType.custom,
                 ['ms-Panel--hasCloseButton ' + styles.rootHasCloseButton]: hasCloseButton
               })
             }
           >
             { overlay }
             <FocusTrapZone
+              ref={ (item) => {
+                if (type === PanelType.custom && item != null) {
+                  item.refs.root.style.width = this.props.customWidth;
```

Comment:
We shouldn't be modifying the DOM like this, or reaching into refs (privates) or assuming root will be there...

instead, couldn't you just pass this in as a style?

```
<FocusTrapZone style={
  (condition) && { width: this.props.customWidth }
}>
```